File: ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  length = str.length
  characters  = str.split("")
  r;
  characters.each do |character|
  r.append(character)
  end
r.join
return r 
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  let reversed = ''

  for(let char of str){
    reversed = char + reversed
  }
return reversed
}

function fibonachi(num){
  let n1 = 0;
  let n2 = 1;
  for (let i= 0; i < num;i++){
    let n3 = n2
    n2 = n1 + n2 
    n1 = n3
  }
  return n1
}
console.log(fibonachi(11))


if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
